packages/shared/useToNumber/index.ts

```diff
@@ -41,6 +41,14 @@ export function useToNumber(
     let resolved = toValue(value)
     if (typeof resolved === 'string')
       resolved = Number[method](resolved, radix)
+    if (!Number.isNaN(resolved)) {
+      if (String(toValue(value)) > String(Number.MAX_SAFE_INTEGER)) {
+        console.warn('useToNumber: value is reaching the limit of Number.MAX_SAFE_INTEGER, resolved value may not be accurate')
+      }
+      else if (String(toValue(value)) < String(Number.MIN_SAFE_INTEGER)) {
+        console.warn('useToNumber: value is reaching the limit of Number.MIN_SAFE_INTEGER, resolved value may not be accurate')
+      }
```

Comment:
How about using `Number.isSafeInteger()` ?

Reply:
Looks good, I didn't even know about this API before. 

However, it seems that there is no way to distinguish whether it exceeds the maximum value or is less than the minimum value. Perhaps a comparison is still needed to give a corresponding warning prompt. What do you think?

Reply:
Indeed there is not, I based my comment on the title of the PR.
Is there a reason you need to know if it's out of  max or min safe integer ?

Reply:
I have encountered similar scenarios, where I need to convert some string IDs into numbers. However, if the ID is out of range, the resulting number is a number with lost precision, which may not be what I expected. If a warning message is given, it can help developers find the problem more clearly and quickly.

Reply:
I mean, is it enough if we just say that it's out of safe boudaries ?

Reply:
I can only give a warning. Do you have any good suggestions?

Reply:
yes :+1: , I think a warning like `useToNumber: ${value} not a safe integer, resolved value may not be accurate`

Reply:
Looks good, updated.